continuous-deployment/readme.md

```diff
@@ -1,7 +1,142 @@
+
+* _An enumeration of the environments available for acceptance, capacity, integration, and user acceptance testing, and the process by which builds will be moved through these environments._
+
+* _A description of the processes to be followed for deployment into testing and production environments, such as change requests to be opened and approvals that need to be granted._
+
+* _A discussion of the method by which the application’s deploy-time and runtime configuration will be managed, and how this relates to the automated deployment process._
+
+* _Description of the integration with any external systems. At what stage and how are they tested as part of a release? How does the technical operator communicate with the provider in the event of a problem?_
+
+* _A disaster recovery plan so that the application’s state can be recovered following a disaster. Which steps will need to be in place to restart or redeploy the application should it fail._
+
+* _Production sizing and capacity planning: How much data will your live application create? How many log files or databases will you need? How much bandwidth and disk space will you need? What latency are clients expecting?_
+
+* _How the initial deployment to production works._
+
+* _How fixing defects and applying patches to the production environment will be handled._
+
+* How upgrades to the production environment will be handled, including data migration. How will upgrades be carried out to the application without destroying its state.
+
+### Application Release and Environment Promotion
+
+Your release manifestation process should take the binaries created from your commit stage and deploy them across all cloud environments, starting with your test environment.
+
+The test environment (_often called Integration_) acts as a gate to validate your test suite completes successfully for all release candidates.
+
+#### The First Deployment
+
+The very first deployment of any application should be showcased to the customer in a production-like environment (_UAT_) to solicit feedback early and often.
+
+#### Criteria for a production-like environment
+
+* Runs the same operating system as production.
+* Has the same software installed as production.
+* Is sized and configured the same way as production.
+* Mirrors production's networking topology.
+* Simulated production-like load tests are executed following a release to surface any latency or throughput degradation.
+
+#### Modeling your Release Pipeline
+
+It's critical to model your test and release process to establish a common understanding between the application engineers and customer stakeholders. Specifically aligning expectations for how many cloud environments need to be pre-provisioned as well as defining sign-off gate roles and responsibilities.
+
+![image](https://user-images.githubusercontent.com/7635865/89290276-426ca200-d61e-11ea-9c79-bd183dc72050.png)
+
+##### The release pipeline model should depict:
```

Comment:
Can be re-written under `Release Pipeline Model` and so each statement is independent of each other. Also a heading with a trailing punctuation should be avoided in markdown as described [here](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md026---trailing-punctuation-in-heading)